// @quillforms/renderer-core/build/components/submit-btn/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _element = require("@wordpress/element");
var _data = require("@wordpress/data");
var _classnames = _interopRequireDefault(require("classnames"));
var _reactLoaderSpinner = require("react-loader-spinner");
var _emotion = require("emotion");
var _button = _interopRequireDefault(require("../button"));
var _htmlParser = _interopRequireDefault(require("../html-parser"));
var _hooks = require("../../hooks");
var _fieldRender = require("../field-render");
/* eslint-disable no-nested-ternary */
/**
 * WordPress Dependencies
 */

/**
 * External Dependencies
 */

/**
 * Internal Dependencies
 */

const SubmitBtn = () => {
  const messages = (0, _hooks.useMessages)();
  const {
    isLastField,
    isActive,
    attributes
  } = (0, _fieldRender.__experimentalUseFieldRenderContext)();
  const theme = (0, _hooks.useBlockTheme)(attributes?.themeId);
  const {
    goToBlock,
    goNext,
    setIsReviewing,
    setIsSubmitting,
    setIsFieldValid,
    setFieldValidationErr,
    setSubmissionErr,
    completeForm,
    setIsCurrentBlockSafeToSwipe,
    setIsFieldPending
  } = (0, _data.useDispatch)('quillForms/renderer-core');
  const {
    onSubmit,
    beforeGoingNext
  } = (0, _hooks.useFormContext)();
  const [isWaitingPending, setIsWaitingPending] = (0, _element.useState)(false);
  const {
    answers,
    firstInvalidFieldId,
    pendingMsg,
    isSubmitting,
    submissionErr,
    currentBlockId
  } = (0, _data.useSelect)(select => {
    return {
      currentBlockId: select('quillForms/renderer-core').getCurrentBlockId(),
      answers: select('quillForms/renderer-core').getAnswers(),
      pendingMsg: select('quillForms/renderer-core').getPendingMsg(),
      isSubmitting: select('quillForms/renderer-core').isSubmitting(),
      firstInvalidFieldId: select('quillForms/renderer-core').getFirstInvalidFieldId(),
      submissionErr: select('quillForms/renderer-core').getSubmissionErr()
    };
  });
  if (!currentBlockId) return null;
  const handleKeyDown = e => {
    if (e.key === 'Enter') {
      if (e.metaKey) {
        submitHandler();
      }
    }
  };
  const goToFirstInvalidField = () => {
    if (firstInvalidFieldId) goToBlock(firstInvalidFieldId);
  };
  (0, _element.useEffect)(() => {
    if (isLastField && isActive) {
      setIsReviewing(false);
      window.addEventListener('keydown', handleKeyDown);
    } else {
      removeEventListener('keydown', handleKeyDown);
    }
    return () => removeEventListener('keydown', handleKeyDown);
  }, [isLastField, isActive]);
  const submitHandler = async () => {
    if (beforeGoingNext && currentBlockId) {
      await beforeGoingNext({
        answers,
        setIsFieldValid,
        setFieldValidationErr,
        goToBlock,
        currentBlockId,
        goNext,
        setIsCurrentBlockSafeToSwipe,
        setIsPending: val => setIsFieldPending(currentBlockId, val)
      });
    }
    if (pendingMsg === false) {
      reviewAndSubmit();
    } else {
      setIsWaitingPending(true);
    }
  };
  const reviewAndSubmit = () => {
    setIsReviewing(false);
    if (firstInvalidFieldId) {
      setTimeout(() => {
        setIsReviewing(true);
      }, 50);
      setTimeout(() => {
        goToFirstInvalidField();
      }, 100);
    } else {
      setIsSubmitting(true);
      onSubmit({
        answers
      }, {
        setIsSubmitting,
        setIsFieldValid,
        setFieldValidationErr,
        setIsReviewing,
        goToBlock,
        completeForm,
        setSubmissionErr,
        setIsPending: val => setIsFieldPending(currentBlockId, val)
      });
    }
  };
  (0, _element.useEffect)(() => {
    if (isWaitingPending && pendingMsg === false) {
      setIsWaitingPending(false);
      reviewAndSubmit();
    }
  }, [isWaitingPending, pendingMsg]);
  return (0, _element.createElement)("div", {
    className: "renderer-core-submit-btn-wrapper"
  }, (0, _element.createElement)(_button.default, {
    className: "renderer-core-submit-btn",
    onClick: () => {
      if (!isSubmitting) submitHandler();
    },
    onKeyDown: e => {
      if (e.key === 'Enter') {
        e.stopPropagation();
        if (!isSubmitting) submitHandler();
      }
    },
    theme: theme
  }, (0, _element.createElement)(_htmlParser.default, {
    value: isWaitingPending ? pendingMsg ? pendingMsg : '' : messages['label.submitBtn']
  }), (isWaitingPending || isSubmitting) && (0, _element.createElement)(_reactLoaderSpinner.TailSpin, {
    wrapperClass: "renderer-core-submit-btn__loader",
    color: "#fff",
    height: 20,
    width: 20
  })), submissionErr && (0, _element.createElement)("div", {
    className: (0, _classnames.default)('renderer-core-submit-error', (0, _emotion.css)`
							color: ${theme.questionsColor};
							margin-top: 15px;
						`)
  }, submissionErr));
};
var _default = SubmitBtn;
exports.default = _default;
