// @quillforms/admin-components/build/rich-text/components/merge-tag-list-item/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _element = require("@wordpress/element");
var _ = require("../../..");
var _classnames = _interopRequireDefault(require("classnames"));
var _emotion = require("emotion");
var _getPlainExcerpt = _interopRequireDefault(require("../../get-plain-excerpt"));
/**
 * QuillForms Dependencies
 */

/**
 * External Dependencies
 */

/**
 * Internal Dependencies
 */

const MergeTagListItem = _ref => {
  let {
    mergeTag,
    onClick,
    onMouseEnter,
    isSelected
  } = _ref;
  return (0, _element.createElement)("div", {
    role: "presentation",
    className: (0, _classnames.default)('rich-text-merge-tag-list-item', {
      isSelected
    }),
    onClick: onClick,
    onMouseEnter: onMouseEnter
  }, (0, _element.createElement)("div", {
    className: (0, _classnames.default)('rich-text-merge-tag-list-item__hidden-bg', (0, _emotion.css)`
						background: ${mergeTag?.color ? mergeTag.color : '#bb426f'};
					`)
  }), (0, _element.createElement)(_.BlockIconBox, {
    color: mergeTag?.color,
    icon: mergeTag?.icon ? mergeTag.icon : null,
    order: mergeTag?.order ? mergeTag.order : null
  }), (0, _element.createElement)("div", {
    className: "rich-text-merge-tag-list-item__title",
    dangerouslySetInnerHTML: {
      __html: mergeTag?.label ? (0, _getPlainExcerpt.default)(mergeTag.label) : ''
    }
  }));
};
var _default = MergeTagListItem;
exports.default = _default;
