// @quillforms/renderer-core/build/components/form-footer/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _element = require("@wordpress/element");
var _data = require("@wordpress/data");
var _classnames = _interopRequireDefault(require("classnames"));
var _fieldNavigation = _interopRequireDefault(require("../field-navigation"));
var _emotion = require("emotion");
var _progressBar = _interopRequireDefault(require("../progress-bar"));
var _hooks = require("../../hooks");
/**
 * WordPress Dependencies
 */

/**
 *
 * External Dependencies
 */

/**
 * Internal Dependencies
 */

const FormFooter = (0, _element.memo)(() => {
  const theme = (0, _hooks.useCurrentTheme)();
  const {
    formObj
  } = (0, _hooks.useFormContext)();
  const {
    currentBlockId,
    isWelcomeScreenActive,
    isThankyouScreenActive,
    shouldFooterBeDisplayed
  } = (0, _data.useSelect)(select => {
    return {
      currentBlockId: select('quillForms/renderer-core').getCurrentBlockId(),
      isThankyouScreenActive: select('quillForms/renderer-core').isThankyouScreenActive(),
      isWelcomeScreenActive: select('quillForms/renderer-core').isWelcomeScreenActive(),
      shouldFooterBeDisplayed: select('quillForms/renderer-core').shouldFooterBeDisplayed()
    };
  });
  if (!currentBlockId) return null;
  return (0, _element.createElement)("div", {
    className: (0, _classnames.default)('renderer-components-form-footer', {
      hidden: isWelcomeScreenActive || isThankyouScreenActive
    }, (0, _emotion.css)`
					@media ( min-width: 768px ) {
						background: ${theme.formFooterBgColor.lg} !important;
					}
					@media ( max-width: 767px ) {
						background: ${theme.formFooterBgColor.sm} !important;
					}
				`),
    tabIndex: -1
  }, !formObj?.settings?.disableProgressBar && (0, _element.createElement)(_progressBar.default, null), !formObj?.settings?.disableNavigationArrows && (0, _element.createElement)(_fieldNavigation.default, {
    shouldFooterBeDisplayed: shouldFooterBeDisplayed
  }));
});
var _default = FormFooter;
exports.default = _default;
