// @quillforms/admin-components/build/addon-feature-availability/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _element = require("@wordpress/element");
var _navigation = require("@quillforms/navigation");
var _config = _interopRequireDefault(require("@quillforms/config"));
var _components = require("@wordpress/components");
var _emotion = require("emotion");
var _lockIcon = _interopRequireDefault(require("./lock-icon"));
/**
 * QuillForms Dependencies
 */

/**
 * WordPress Dependencies
 */

/**
 * External Dependencies
 */

/** 
 * Internal Dependencies
 */

const AddonFeatureAvailability = _ref => {
  let {
    featureName,
    addonSlug,
    showLockIcon
  } = _ref;
  const license = _config.default.getLicense();
  const addon = _config.default.getStoreAddons()[addonSlug];
  const featurePlanLabel = _config.default.getPlans()[addon.plan].label;
  const isPlanAccessible = _config.default.isPlanAccessible(addon.plan);
  let content = (0, _element.createElement)("div", null);

  // if addon installed.
  if (addon.is_installed) {
    // license note in case of invalid or low level license.
    let licenseNote = (0, _element.createElement)("div", null);
    if (license?.status !== 'valid') {
      licenseNote = (0, _element.createElement)("div", {
        className: "addon-feature-availability-license-note"
      }, "Renewing your license is recommended to receive updates for this addon.");
    } else if (!isPlanAccessible) {
      licenseNote = (0, _element.createElement)("div", {
        className: "addon-feature-availability-license-note"
      }, "Upgrading your license to ", featurePlanLabel, " is recommended to receive updates for this addon.");
    }
    // if addon is active, shouldn't be reached in most cases.
    if (addon.is_active) {
      content = (0, _element.createElement)("div", null, (0, _element.createElement)("div", null, featureName, " is already available."), licenseNote);
      // else, the addon is not active -but installed-.
    } else {
      content = (0, _element.createElement)("div", null, (0, _element.createElement)("div", null, featureName, " is already available."), (0, _element.createElement)("div", null, "Please activate ", (0, _element.createElement)("b", null, addon.name), " addon from", ' ', (0, _element.createElement)(_navigation.NavLink, {
        to: `/admin.php?page=quillforms&path=addons`
      }, "addons page"), ' ', "to use it."), licenseNote);
    }
    // else, addon is not installed.
  } else {
    // if license is valid and the feature plan is accessible.
    if (license?.status === 'valid' && isPlanAccessible) {
      content = (0, _element.createElement)("div", null, (0, _element.createElement)("div", null, featureName, " is available for your plan."), (0, _element.createElement)("div", null, "Please install and activate ", addon.name, " addon from", ' ', (0, _element.createElement)(_navigation.NavLink, {
        to: `/admin.php?page=quillforms&path=addons`
      }, "addons page"), ' ', "to use it."));
      // else, no license or the license is invalid or the feature of addon require higher plan.
    } else {
      content = (0, _element.createElement)("div", null, showLockIcon && (0, _element.createElement)(_components.Icon, {
        className: "addon-feature-availability-lock-icon",
        icon: _lockIcon.default,
        size: 120
      }), (0, _element.createElement)("p", {
        className: (0, _emotion.css)`
							font-size: 15px;
						`
      }, "We're sorry, ", featureName, " is not available", (0, _element.createElement)("br", null), "on your plan. Please upgrade to the ", featurePlanLabel, ' ', "plan to unlock", (0, _element.createElement)("br", null), "all of ", featurePlanLabel, " features."), (0, _element.createElement)("a", {
        href: "https://quillforms.com",
        target: "_blank",
        className: "addon-feature-availability-upgrade-button"
      }, "Upgrade to ", featurePlanLabel, "!"));
    }
  }
  return (0, _element.createElement)("div", {
    className: "addon-feature-availability"
  }, content);
};
var _default = AddonFeatureAvailability;
exports.default = _default;
