// @quillforms/admin-components/build/rich-text/components/hovering-toolbar/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _element = require("@wordpress/element");
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = require("react");
var _reactDom = require("react-dom");
var _emotion = require("emotion");
var _slateReact = require("slate-react");
var _slate = require("slate");
var _link = _interopRequireDefault(require("../link"));
var _formatButton = _interopRequireDefault(require("../format-button"));
var _boldIcon = _interopRequireDefault(require("./bold-icon"));
var _italicIcon = _interopRequireDefault(require("./italic-icon"));
/**
 * WordPress Dependencies
 */

/**
 * External Dependencies
 */

/**
 * Internal Dependencies
 */

const Menu = (0, _react.forwardRef)((_ref, ref) => {
  let {
    className,
    ...props
  } = _ref;
  return (0, _element.createElement)("div", (0, _extends2.default)({}, props, {
    ref: ref,
    className: (0, _emotion.cx)(className, (0, _emotion.css)`
					& > * {
						display: inline-block;
					}
					& > * + * {
						margin-left: 15px;
					}
				`)
  }));
});
const HoveringToolbar = _ref2 => {
  let {
    toggleFormat,
    isFormatActive,
    formattingControls
  } = _ref2;
  const ref = (0, _react.useRef)(null);
  const editor = (0, _slateReact.useSlate)();
  const inFocus = (0, _slateReact.useFocused)();
  (0, _react.useEffect)(() => {
    const el = ref.current;
    const {
      selection
    } = editor;
    if (!el) {
      return;
    }
    if (!selection || !inFocus || _slate.Range.isCollapsed(selection) || _slate.Editor.string(editor, selection) === '') {
      el.removeAttribute('style');
      return;
    }
    const domSelection = window.getSelection();
    const domRange = domSelection?.getRangeAt(0);
    const rect = domRange?.getBoundingClientRect();
    el.style.opacity = '1';
    if (rect) {
      el.style.top = `${rect.top + window.pageYOffset - el.offsetHeight}px`;
      el.style.left = `${rect.left + window.pageXOffset - el.offsetWidth / 2 + rect.width / 2}px`;
    }
  });
  return (0, _reactDom.createPortal)((0, _element.createElement)(Menu, {
    ref: ref,
    className: (0, _emotion.css)`
				padding: 8px 7px 6px;
				position: absolute;
				z-index: 1;
				top: -10000px;
				left: -10000px;
				margin-top: -6px;
				opacity: 0;
				background-color: #222;
				border-radius: 4px;
				transition: opacity 0.75s;

				svg {
					width: 20px;
				}
			`
  }, !!formattingControls && formattingControls.includes('bold') && (0, _element.createElement)(_formatButton.default, {
    format: "bold",
    toggleFormat: () => toggleFormat('bold'),
    isFormatActive: () => isFormatActive('bold')
  }, (0, _element.createElement)(_boldIcon.default, null)), !!formattingControls && formattingControls.includes('italic') && (0, _element.createElement)(_formatButton.default, {
    format: "italic",
    toggleFormat: () => toggleFormat('italic'),
    isFormatActive: () => isFormatActive('italic')
  }, (0, _element.createElement)(_italicIcon.default, null)), !!formattingControls && formattingControls.includes('link') && (0, _element.createElement)(_link.default, {
    editor: editor
  })), document.body);
};
var _default = HoveringToolbar;
exports.default = _default;
