// @quillforms/admin-components/build/rich-text/components/element/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _element = require("@wordpress/element");
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _slateReact = require("slate-react");
var _mergeTag = _interopRequireDefault(require("../merge-tag"));
/**
 * External Dependencies
 */

/**
 * Internal Dependencies
 */

const Element = _ref => {
  let {
    attributes,
    children,
    element,
    editor,
    mergeTags
  } = _ref;
  const path = _slateReact.ReactEditor.findPath(editor, element);
  if (element.type === 'mergeTag') {
    return (0, _element.createElement)(_mergeTag.default, {
      element: element,
      mergeTags: mergeTags,
      editor: editor,
      path: path,
      attributes: attributes,
      children: children
    });
  } else if (element.type === 'link') {
    return (0, _element.createElement)("a", (0, _extends2.default)({}, attributes, {
      href: element.url
    }), children);
  }
  return (0, _element.createElement)("p", attributes, children);
};
var _default = Element;
exports.default = _default;
